components/code-editor/index.js

```diff
@@ -102,4 +183,29 @@ class CodeEditor extends Component {
 	}
 }
 
-export default CodeEditor;
+export default withLazyDependencies( {
+	scripts: [
+		'wp-codemirror',
+		'code-editor',
+		'htmlhint',
+		'csslint',
+		'jshint',
+		// TODO: Gotta check if user can unfiltered_html
+		...( true ? [ 'htmlhint-kses' ] : [] ),
```

Comment:
You can always give an option to pass object or function to this HOC. When there is a function you can call it with all props passed to the component and based on this perform required filtering.